src/mappings/horizonStaking.ts

```diff
@@ -165,11 +165,67 @@ export function handleTokensDeprovisioned(event: TokensDeprovisioned): void {
 }
 
 export function handleProvisionParametersSet(event: ProvisionParametersSet): void {
-    // To Do
+    let provision = createOrLoadProvision(event.params.serviceProvider, event.params.verifier, event.block.timestamp)
+    provision.thawingPeriod = event.params.thawingPeriod
+    provision.maxVerifierCut = event.params.maxVerifierCut
+    provision.save()
 }
 
 export function handleProvisionParametersStaged(event: ProvisionParametersStaged): void {
-    // To Do
+    let provision = createOrLoadProvision(event.params.serviceProvider, event.params.verifier, event.block.timestamp)
+    provision.thawingPeriodPending = event.params.thawingPeriod
+    provision.maxVerifierCutPending = event.params.maxVerifierCut
+    provision.save()
+}
+
+export function handleOperatorSet(event: OperatorSet): void {
+    let indexerGraphAccount = createOrLoadGraphAccount(event.params.serviceProvider, event.block.timestamp)
+    let operator = createOrLoadOperator(event.params.operator, event.params.verifier, event.params.serviceProvider)
+    let operators = indexerGraphAccount.operators
+    // Will have to handle legacy operators list, and horizon provisionedOperators list for extra context
```

Comment:
Why cant we deprecate and remove the legacy operators list? Just process event history and add to the new operator list?

Reply:
I think it was for backwards compatibility and also the new operator list is for operator <-> Provision relationship rather than Operator <-> Indexer relationship, but I don't really remember now, will check soon

Reply:
Yup, essentially the new HorizonOperator needs to know it's provision. We could make it optional and add everything to the same operator list, but... that unnecessarily breaks compatibility for the old stuff.

We could do it so that everything is in the same place, but given we will have a Stage 2 subgraph that essentially erases all pre-Horizon data, having a clear cut as to how to support each side, while still making it compatible with the legacy way (but only for legacy data) seemed to make the most sense